test/framework/framework.go

```diff
@@ -577,16 +577,24 @@ func (f *Framework) DeletePrometheusOperatorClusterResource(ctx context.Context)
 }
 
 func (f *Framework) SetupPrometheusRBAC(ctx context.Context, t *testing.T, testCtx *TestCtx, ns string) {
-	if _, err := f.CreateOrUpdateClusterRole(ctx, fmt.Sprintf("%s/rbac/prometheus/prometheus-cluster-role.yaml", f.exampleDir)); err != nil && !apierrors.IsAlreadyExists(err) {
-		t.Fatalf("failed to create or update prometheus cluster role: %v", err)
-	}
 	if finalizerFn, err := f.createOrUpdateServiceAccount(ctx, ns, fmt.Sprintf("%s/rbac/prometheus/prometheus-service-account.yaml", f.exampleDir)); err != nil {
 		t.Fatal(errors.Wrap(err, "failed to create or update prometheus service account"))
 	} else {
 		if testCtx != nil {
 			testCtx.AddFinalizerFn(finalizerFn)
 		}
+	}
 
+	if _, err := f.CreateOrUpdateClusterRole(ctx, fmt.Sprintf("%s/rbac/prometheus/prometheus-cluster-role.yaml", f.exampleDir)); err != nil && !apierrors.IsAlreadyExists(err) {
+		t.Fatalf("failed to create or update prometheus cluster role: %v", err)
+	}
+
+	if finalizerFn, err := f.createOrUpdateClusterRoleBinding(ctx, ns, fmt.Sprintf("%s/rbac/prometheus/prometheus-cluster-role-binding.yaml", f.exampleDir)); err != nil {
```

Comment:
Do we need to add a deletion step in the function `DeletePrometheusOperatorClusterResource` to delete the ClusterRoleBinding?

I noticed we don't have test failures in the PR so I'm not sure, but I'd expect that we would need to delete it 🤔

Reply:
Ya I see that we could delete it. I just followed similar to RoleBinding :)